internal/provider/utils/testutils.go

```diff
@@ -131,3 +135,318 @@ func CACerts(t *testing.T) (string, string) {
 
 	return string(pem.EncodeToMemory(root)), string(pem.EncodeToMemory(intermediate))
 }
+
+func NewWebhook(t *testing.T, provisionerID, authorityID string) *v20230301.ProvisionerWebhook {
+	client, err := SmallstepAPIClientFromEnv()
+	if err != nil {
+		t.Fatal(err)
+	}
+	url := "https://example.com/hook"
+	name, err := randutil.Alphanumeric(8)
+	require.NoError(t, err)
+
+	req := v20230301.ProvisionerWebhook{
+		Name:       name,
+		Url:        &url,
+		Kind:       "ENRICHING",
+		CertType:   "ALL",
+		ServerType: "EXTERNAL",
+	}
+
+	resp, err := client.PostWebhooks(context.Background(), authorityID, provisionerID, &v20230301.PostWebhooksParams{}, req)
+	require.NoError(t, err)
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	require.NoError(t, err)
+
+	require.Equal(t, http.StatusCreated, resp.StatusCode, "got %d: %s", resp.StatusCode, body)
+
+	wh := &v20230301.ProvisionerWebhook{}
+	require.NoError(t, json.Unmarshal(body, wh))
+	wh.Secret = nil
+
+	return wh
+}
+
+func NewCollection(t *testing.T) *v20230301.Collection {
+	client, err := SmallstepAPIClientFromEnv()
+	require.NoError(t, err)
+
+	slug := Slug(t)
+	displayName := "Collection " + slug
+
+	req := v20230301.NewCollection{
+		Slug:        slug,
+		DisplayName: &displayName,
+	}
+
+	resp, err := client.PostCollections(context.Background(), &v20230301.PostCollectionsParams{}, req)
+	require.NoError(t, err)
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	require.NoError(t, err)
+
+	require.Equal(t, http.StatusCreated, resp.StatusCode, "got %d: %s", resp.StatusCode, body)
+
+	collection := &v20230301.Collection{}
+	err = json.Unmarshal(body, collection)
+	require.NoError(t, err)
+
+	return collection
+}
+
+func NewInstance(t *testing.T, slug string) *v20230301.CollectionInstance {
```

Comment:
`NewInstance` -> `NewCollectionInstance` for easier disambiguation?

Reply:
0ac66ff